pkg/shp/flags/flags.go

```diff
@@ -173,35 +231,32 @@ func timeoutFlags(flags *pflag.FlagSet, timeout *metav1.Duration) {
 }
 
 // buildRefFlags register flags for BuildRun's spec.buildRef attribute.
-func buildRefFlags(flags *pflag.FlagSet, buildRef *buildv1alpha1.BuildRef) {
+func buildRefFlags(flags *pflag.FlagSet, buildRef *buildv1beta1.ReferencedBuild) {
 	flags.StringVar(
-		&buildRef.Name,
+		buildRef.Name,
 		BuildrefNameFlag,
 		"",
 		"name of build resource to reference",
 	)
-	flags.StringVar(
-		buildRef.APIVersion,
-		"buildref-apiversion",
-		"",
-		"API version of build resource to reference",
-	)
 }
 
 // serviceAccountFlags register flags for BuildRun's spec.serviceAccount attribute.
-func serviceAccountFlags(flags *pflag.FlagSet, sa *buildv1alpha1.ServiceAccount) {
+func serviceAccountFlags(flags *pflag.FlagSet, sa *string) {
 	flags.StringVar(
-		sa.Name,
+		sa,
 		ServiceAccountNameFlag,
 		"",
 		"Kubernetes service-account name",
 	)
+	var ignore bool
 	flags.BoolVar(
-		sa.Generate,
+		&ignore,
 		ServiceAccountGenerateFlag,
 		false,
 		"generate a Kubernetes service-account for the build",
 	)
+	flags.MarkDeprecated("sa-generate", fmt.Sprintf("this flag has no effect, please use --%s for service account", ServiceAccountNameFlag))
```

Comment:
Note that with this change, we should also remove the service account generation logic in the BuildRun controller. Filed https://github.com/shipwright-io/build/issues/1873

Reply:
So, we don't need any change in cli. Right ?